iac/terraform/aws/cognito/cognito.tf

```diff
@@ -57,6 +57,35 @@ resource "aws_cognito_user_pool" "user_pool" {
   }
 }
 
+resource "aws_cognito_user_pool_domain" "set_custom_domain_prod" {
+  count = var.environment == module.environments.PRODUCTION ? 1 : 0
+
+  domain          = "${var.bloodconnect_domain}"
+  user_pool_id    = aws_cognito_user_pool.user_pool.id
+  certificate_arn = var.acm_certificate_arn
+}
+
+resource "aws_cognito_user_pool_domain" "set_custom_domain" {
+  count = var.environment == module.environments.PRODUCTION ? 0 : 1
+
+  domain          = replace(replace(replace(replace("${var.environment}", "aws", ""), "amazon", ""), "cognito", ""), "-", "")
+  user_pool_id    = aws_cognito_user_pool.user_pool.id
+}
+
+resource "aws_route53_record" "cognito_user_pool_custom_domain" {
+  count = var.environment == module.environments.PRODUCTION ? 1 : 0
+
+  name    = aws_cognito_user_pool_domain.set_custom_domain_prod[0].domain
+  type    = "A"
```

Comment:
are you sure its a A type? its not an ip address, A records are IP addresses

Reply:
Yes , according to the [Custom Cognito domain](https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/cognito_user_pool_domain#custom-cognito-domain)

Reply:
the link doesn't tell me anything

Reply:
i feel like this will be a CNAME

Reply:
Theoretically it should be  `CNAME`  as it's not directly pointing to any `IP address`. I'll check if any issue occurs or not.